* src/su.c: shell's name must be -su when a su fakes a login.

===================================================================
--- src/su.c	(revision 3193)
+++ src/su.c	(revision 3194)
@@ -1021,7 +1021,7 @@
 		 * Use the shell and create an argv
 		 * with the rest of the command line included.
 		 */
-		argv[-1] = shellstr;
+		argv[-1] = cp;
 #ifndef USE_PAM
 		execve_shell (shellstr, &argv[-1], environ);
 		err = errno;
 
